mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1002,10 +1008,14 @@ private[spark] object RandomForest extends Logging {
       val numSplits = metadata.numSplits(featureIndex)
 
       // get count for each distinct value
-      val (valueCountMap, numSamples) = featureSamples.foldLeft((Map.empty[Double, Int], 0)) {
+      var (valueCountMap, numSamples) = featureSamples.foldLeft((Map.empty[Double, Int], 0)) {
```

Comment:
I'd probably avoid a `var` by using a new variable for `numSamples` here.

Reply:
problem has been solved